### PR TITLE
chore(3000): remove legacy top bar

### PR DESCRIPTION
**Pull request by thmsobrmlr**
## Problem

Most of the `<TopBar />` component is not used in 3000 any more.

## Changes

This PR removes the component and associated files. Still used things are left in place of course. In a follow-up the navigation-3000 components are going to be moved to the navigation folder.

While working on this PR I realized that we haven't implemented cloud-announcements for 3000 and did not refactor all places of navigationLogic / activationLogic with now unused code, as I'd like the growth team to review the changes there as well.

Relevant chores were added to https://github.com/PostHog/posthog/pull/19416.

## How did you test this code?

Re-built the app and verified the 3000 UI works as expected

### CODE REVIEW

File: frontend/__snapshots__/layout-navigation--app-page-with-side-bar-hidden--dark.png
Comment (thmsobrmlr) [inline frontend/__snapshots__/layout-navigation--app-page-with-side-bar-hidden--dark.png:1]: These are snapshots for the old navigation component that are going to be removed with the rest of the component.

File: frontend/__snapshots__/lemon-ui-icons--shelf-m--dark.png
Comment (thmsobrmlr) [inline frontend/__snapshots__/lemon-ui-icons--shelf-m--dark.png:1]: We aren't using `<IconMenuOpen />` any more.

File: frontend/src/layout/navigation/TopBar/Announcement.tsx [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
 import { useActions, useValues } from 'kea'
-import { MOCK_NODE_PROCESS } from 'lib/constants'
 import { NewFeatureBanner } from 'lib/introductions/NewFeatureBanner'
 import { IconClose } from 'lib/lemon-ui/icons'
 import { LemonMarkdown } from 'lib/lemon-ui/LemonMarkdown'
 import { preflightLogic } from 'scenes/PreflightCheck/preflightLogic'
 
 import { announcementLogic, AnnouncementType } from '~/layout/navigation/TopBar/announcementLogic'
 
-window.process = MOCK_NODE_PROCESS
```
Comment (thmsobrmlr) [inline frontend/src/layout/navigation/TopBar/Announcement.tsx:12]: This is now handled in `frontend/src/scenes/App.tsx`.

File: playwright/e2e-vrt/layout/Navigation.spec.ts-snapshots/Navigation-App-Page-With-Side-Bar-Shown-Mobile-1-chromium-linux.png
Comment (daibhin) [inline playwright/e2e-vrt/layout/Navigation.spec.ts-snapshots/Navigation-App-Page-With-Side-Bar-Shown-Mobile-1-chromium-linux.png:1]: Assuming these snapshots will be removed at some point too?

File: frontend/src/lib/lemon-ui/icons/icons.tsx [hunk 2]
```diff
@@ -512,18 +512,6 @@ export function IconMenu(props: LemonIconProps): JSX.Element {
     )
 }
 
-/** Material Design Menu Open icon. */
-export function IconMenuOpen(props: LemonIconProps): JSX.Element {
```
Comment (daibhin) [inline frontend/src/lib/lemon-ui/icons/icons.tsx:516]: We should remove this icon from https://github.com/posthog/icons if it has already been added & we don't plan on using it again

File: frontend/src/layout/navigation/Navigation.stories.tsx [hunk 3]
```diff
@@ -18,74 +14,57 @@ const meta: Meta = {
...
-
-    useEffect(() => {
-        toggleSideBarBase(false)
-        toggleSideBarMobile(false)
-    }, [])
-
     return <BaseAppPage />
 }
 
 export function AppPageWithSideBarShown(): JSX.Element {
```
Comment (daibhin) [inline frontend/src/layout/navigation/Navigation.stories.tsx:68]: This story and the one above look to do the same thing. Should we consolidate or remove them if no longer necessary?

File: frontend/src/layout/navigation/navigationLogic.ts [hunk 4]
```diff
@@ -61,18 +59,11 @@ export const navigationLogic = kea<navigationLogicType>([
...
     reducers({
         // Non-mobile base
-        isSideBarShownBase: [
-            true,
-            { persist: true },
-            {
-                toggleSideBarBase: (state, { override }) => override ?? !state,
-            },
-        ],
+        isSideBarShownBase: [true, { persist: true }, {}],
```
Comment (daibhin) [inline frontend/src/layout/navigation/navigationLogic.ts:62]: Can this be removed? Looks like it is not referenced anywhere in the codebase

File: frontend/src/layout/navigation/TopBar/Announcement.scss [hunk 5]
```diff
@@ -0,0 +1,51 @@
+@import '../../../styles/mixins';
```
Comment (daibhin) [inline frontend/src/layout/navigation/TopBar/Announcement.scss:1]: Would be nice to get a story to see what these changes look like / prevent future regressions

**Review by daibhin — APPROVED** [review]
Left a couple of comments but nothing really blocking

File: playwright/e2e-vrt/layout/Navigation.spec.ts-snapshots/Navigation-App-Page-With-Side-Bar-Shown-Mobile-1-chromium-linux.png
Comment (daibhin) [reply playwright/e2e-vrt/layout/Navigation.spec.ts-snapshots/Navigation-App-Page-With-Side-Bar-Shown-Mobile-1-chromium-linux.png:1]: I think this is addressed in https://github.com/PostHog/posthog/pull/19525#discussion_r1437517987

File: frontend/src/lib/lemon-ui/icons/icons.tsx (hunk 2 above)
Comment (thmsobrmlr) [reply frontend/src/lib/lemon-ui/icons/icons.tsx:516]: Yup, it doesn't seem to have been added there

File: frontend/src/layout/navigation/navigationLogic.ts (hunk 4 above)
Comment (thmsobrmlr) [reply frontend/src/layout/navigation/navigationLogic.ts:62]: Yup, I'll remove it with rest of the sidebar code

File: frontend/src/layout/navigation/TopBar/Announcement.scss (hunk 5 above)
Comment (thmsobrmlr) [reply frontend/src/layout/navigation/TopBar/Announcement.scss:1]: We need to add the cloud announcements feature in 3000 and can add the stories there as well

**Pull request merged**